r-package/grf/R/causal_forest.R

```diff
@@ -131,13 +134,14 @@ causal_forest <- function(X, Y, W,
     seed <- validate_seed(seed)
     clusters <- validate_clusters(clusters, X)
     samples_per_cluster <- validate_samples_per_cluster(samples_per_cluster, clusters)
+    honesty.fraction <- validate_honesty_fraction(honesty.fraction, honesty)
     
     reduced.form.weight <- 0
 
     if (is.null(Y.hat)) {
       forest.Y <- regression_forest(X, Y, sample.fraction = sample.fraction, mtry = mtry, tune.parameters = tune.parameters,
-                                    num.trees = min(500, num.trees), num.threads = num.threads, min.node.size = NULL, 
-                                    honesty = TRUE, seed = seed, ci.group.size = 1, alpha = alpha, imbalance.penalty = imbalance.penalty,
+                                    num.trees = min(500, num.trees), num.threads = num.threads, min.node.size = NULL, honesty = honesty,
```

Comment:
Here, we're changing some behavior that is unrelated to the addition of `honesty.fraction`: before the orthogonalization forests were always trained with `honesty`, and now they inherit the passed-in value of `honesty`. In general, it's best not to tack on a non-core change to a PR (especially one that affects user results), as it makes it hard to track down when/ why a change was made.

If we feel that the orthogonalization forests should inherit the `honesty` parameter, I think we should pull this out and commit it in a separate PR.